Source/Scene/Points3DTileContent.js

```diff
@@ -164,159 +196,366 @@ define([
+                    componentsPerAttribute : 3,
+                    componentDatatype : ComponentDatatype.FLOAT,
+                    normalize : false,
+                    offsetInBytes : 0,
+                    strideInBytes : 0
+                });
+            }
+        }
+
+        var vertexArray = new VertexArray({
+            context : context,
+            attributes : attributes
         });
 
-        var primitive = new Primitive({
-            geometryInstances : instance,
-            appearance : appearance,
-            asynchronous : false,
-            allowPicking : false,
-            cull : false,
-            rtcCenter : boundingSphere.center
+        var attributeLocations = {
+            a_position : positionAttributeLocation
+        };
+
+        if (hasColors) {
+            attributeLocations = combine(attributeLocations, {
+                a_color : colorAttributeLocation
+            });
+        }
+
+        if (hasNormals) {
+            attributeLocations = combine(attributeLocations, {
+                a_normal : normalAttributeLocation
+            });
+        }
+
+        var shaderProgram = ShaderProgram.fromCache({
+            context : context,
+            vertexShaderSource : vs,
+            fragmentShaderSource : fs,
+            attributeLocations : attributeLocations
         });
 
-        this._primitive = primitive;
-        this.state = Cesium3DTileContentState.PROCESSING;
-        this.contentReadyToProcessPromise.resolve(this);
+        content._opaqueRenderState = RenderState.fromCache({
+            depthTest : {
+                enabled : true
+            }
+        });
 
-        var that = this;
+        content._translucentRenderState = RenderState.fromCache({
+            depthTest : {
+                enabled : true
+            },
+            depthMask : false,
+            blending : BlendingState.ALPHA_BLEND
+        });
 
-        when(primitive.readyPromise).then(function(primitive) {
-            that.state = Cesium3DTileContentState.READY;
-            that.readyPromise.resolve(that);
-        }).otherwise(function(error) {
-            that.state = Cesium3DTileContentState.FAILED;
-            that.readyPromise.reject(error);
+        content._drawCommand = new DrawCommand({
+            boundingVolume : content._tile.contentBoundingVolume.boundingSphere,
+            cull : false, // Already culled by 3D tiles
+            modelMatrix : new Matrix4(),
+            primitiveType : PrimitiveType.POINTS,
+            vertexArray : vertexArray,
+            count : pointsLength,
+            shaderProgram : shaderProgram,
+            uniformMap : uniformMap,
+            renderState : isTranslucent ? content._translucentRenderState : content._opaqueRenderState,
+            pass : isTranslucent ? Pass.TRANSLUCENT : Pass.OPAQUE,
+            owner : content
         });
-    };
+
+        content.state = Cesium3DTileContentState.READY;
```

Comment:
Minor, but I think it would be cleaner to move these instead the `if` statement in `update` so this function is more cohesive as "create resources", not "create resources and set a bunch of state."